src/skan/csr.py

```diff
@@ -1092,3 +1094,121 @@ def make_degree_image(skeleton_image):
                     bool_skeleton.astype(int), degree_kernel, mode='constant'
                     )
     return degree_image
+
+
+def sholl_analysis(
+        skeleton,
+        soma=None,
+        step_size=None,
+        num_shells=None,
+        start_radius=0,
+        end_radius=None
+        ):
+    """Sholl Analysis for Skeleton object.
+
+    Parameters
+    ----------
+    skeleton : skan.csr.Skeleton
+        A Skeleton object.
+    soma : array-like of float or None, optional
+        Pixel, not real, coordinates of point on skeleton keeping which as
+        center the concentric shells are computed. If None, pixel nearest to
+        geodesic center of skeleton is chosen, by default None
+    step_size : float or None, optional
+        Spacing between intermediate shells. If None, `num_shells` is used. It
+        takes precendence over `num_shells`, by default None
+    num_shells : int, optional
+        Number of concentric shells, by default None
+    start_radius : float, optional
+        The real world radius of the smallest shell, i.e., the first distance
+        to be sampled, by default 0
+    end_radius : float or None, optional
+        The real world radius of the largest (last) shell. If None, it is
+        automatically calculated as largest possible radius, by default None
```

Comment:
I think for the parameters I would choose *either* the full shell spec, which can be easily obtained with `np.arange(start, stop, step)`, *or* `num_shells`, at which point it would be computed from the radius. How about this for an API:

`shells=10`: If an int, divide the dataset radius into this many shells. If an array of floats, use it directly as the different shell radii.

What do you think?

Reply:
I was just modeling it on the ImageJ Sholl Analysis plugin. But yeah, that could make the API unambiguous

Reply:
But consider this corner case if `start_radius` & `end_radius` conflicting it, the array of floats in shells has minimum other than the `start_radius` (maybe lower), or it has maximum that's higher than `end_radius`.

Reply:
Or maybe we can remove the `start_radius` & `end_radius` arguments altogether

Reply:
I did it by removing these arguments